com.unity.render-pipelines.universal/Documentation~/renderer-features/how-to-custom-effect-render-objects.md

```diff
@@ -0,0 +1,129 @@
+
+### Create a Renderer Feature to draw the character behind GameObjects
+
+Follow these steps to create a Renderer Feature to draw the character behind GameObjects.
+
+1. Select a URP Renderer.
+
+    ![Select a URP Renderer](../Images/how-to-render-objects/rendobj-select-urp-renderer.png)
+
+2. In the Inspector, click **Add Renderer Feature** and select **Render Objects**.
+
+    ![Add Render Object Renderer Feature](../Images/how-to-render-objects/rendobj-add-rend-obj.png)
+
+    Call the new Renderer Feature **DrawCharacterBehind**.
+
+3. This example uses Layers to filter the GameObjects to render. Create a new Layer and call it **Character**.
+
+    ![Create new Layer called Character](../Images/how-to-render-objects/rendobj-new-layer-character.png)
+
+4. In the `DrawCharacterBehind` Renderer Feature, in **Filters** > **Layer Mask**, select `Character`. With this setting, this Renderer Feature renders GameObjects only in the Layer `Character`. 
+
+5. In **Overrides** > **Material**, select the `CharacterBehindObjects` Material.
+
+    The Renderer Feature overrides the Material of a GameObject with the selected Material.
+
+    ![Layer Mask, Material Override](../Images/how-to-render-objects/rendobj-change-layer-override-material.png)
+
+6. The intended behavior is that the Renderer Feature renders the character with the `CharacterBehindObjects` Material only when the character is behind other GameObjects.
+
+    To achieve this, select the **Depth** check box, and set the **Depth Test** property to **Greater**. 
+
+    ![Set Depth Test to Greater](../Images/how-to-render-objects/rendobj-depth-greater.png)
+
+With these settings, Unity renders the character with the `CharacterBehindObjects` Material only when the character is behind another GameObject. However, Unity also renders parts of the character using the `CharacterBehindObjects` Material, because some parts of the character occlude the character itself.    
+
+![Unity renders parts of the character using the `CharacterBehindObjects` Material](../Images/how-to-render-objects/character-depth-test-greater.gif)
+
+
+### Create an extra Renderer Feature to avoid the self see-through effect
+
+The settings in the previous section result in the self see-through effect for the following reason:
+
+* When performing the Opaque rendering pass of the URP Renderer, Unity renders all GameObjects belonging to the character with the `Character` Material and writes depth values to the Depth buffer.
+
+* When executing the `DrawCharacterBehind` Renderer Feature, Unity performs the depth test using the condition specified in the **Depth Test** property. In the following screenshot, a bigger capsule occludes part of the smaller capsule, and the depth test passes for that part of the smaller capsule. The Renderer Feature overrides the Material for that part.
```

Comment:
`In the following screenshot, a bigger capsule occludes part of the smaller capsule, and the depth test passes for that part of the smaller capsule. The Renderer Feature overrides the Material for that part.`
One important piece of information that's implied here but maybe could be explicit. 

This happen because the Lit material (red) draws before the DrawCharacterBehind material. 
So by the time the DrawCharacterMaterial is draw, the bigger capsule is already written to depth buffer and then it self occlusion. 

The `render pass event` is what controls the order that these objects are drawn. 
By default new RenderObjects features are draw "AfterOpaques". 

In the next section when you explain how to fix the issue you are removing the `Character` from the opaque layer mask and then adding a renderer feature to draw it with render event "AfterOpaque". 

So essentially that's what changes the order or rendering. 

Before the order was:
1) Character is drawn as part of the renderer in the opaque layer mask (render pass event `BeforeOpaques`)
2) RenderObject "DrawCharacterBehind" draws unlit blue material. (render pass event `AfterOpaques`)

After the fix the order is:
1) Character is skips rendering in `BeforeOpaques`. 
2) CharacterBehind draws in `AfterOpaques`
3) Character drawsin `AfterOpaques` (but after 2)

Reply:
Thanks! I've made changes to this section in the new commit, I'll consider explaining this (the order of Renderer Feature execution) in more detail on a separate page. Jonas also mentioned something similar.